src/qtils/strict_sptr.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <stdexcept>
+#include <type_traits>
+
+namespace qtils {
+
+  /**
+   * @brief A strict version of std::shared_ptr that guarantees non-null values.
+   *
+   * This class prevents default construction and ensures that the shared
+   * pointer is never null. It removes unsafe methods such as `reset()` and
+   * `operator=`. Provides safe access to the underlying std::shared_ptr.
+   *
+   * @tparam T The type of the managed object.
+   */
+  template <class T>
```

Comment:
It would actually be nice to support custom Deleters, but I think this feature may be added later when we need it somewhere.